src/NHibernate/Dialect/DB2Dialect.cs

```diff
@@ -128,8 +114,6 @@ public DB2Dialect()
 
 			// DB2 does not support ANSI substring syntax.
 			RegisterFunction("substring", new SQLFunctionTemplate(NHibernateUtil.String, "substring(?1, ?2, ?3)"));
-
-			DefaultProperties[Environment.ConnectionDriver] = "NHibernate.Driver.DB2Driver";
```

Comment:
Removing this is a breaking change for all apps which were using the dialect without specifying the driver. Better keep-it, eventually guarded by a compilation directive for not specifying an invalid driver under .Net Core and a maybe invalid one under .Net Standard.

Reply:
Reverted